src/ClientRepository.php

```diff
@@ -47,6 +49,20 @@ public function findForUser($clientId, $userId)
                     ->first();
     }
 
+
+    /**
+     * Get a client instance for the given bearer token.
+     *
+     * @param $bearerToken
+     * @return \Laravel\Passport\Client|null
+     */
+    public function findByToken($bearerToken)
+    {
+        $clientId = (new Parser())->parse($bearerToken)->getClaim('aud');
+
+        return $this->find($clientId);
+    }
```

Comment:
I don't think it's a good idea to move this logic to the `ClientRepository`. You're mixing parsing tokens with retrieving clients which are two separate things. The `ClientRepository` only job is to retrieve client models.

A better place for this method would be the `TokenGuard` class. Try to see if you can implement a `client` method like the `user` method:

```
    /**
     * Get the user for the incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return mixed
     */
    public function user(Request $request)
    {
        if ($request->bearerToken()) {
            return $this->authenticateViaBearerToken($request);
        } elseif ($request->cookie(Passport::cookie())) {
            return $this->authenticateViaCookie($request);
        }
    }
```